app/pdf_redactor.py

```diff
@@ -700,67 +588,79 @@ def update_text_layer(options, text_tokens, page_tokens):
 	if len(text_tokens) == 0:
 		# No text content.
 		return
-
 	# Apply each regular expression to the text content...
 	for pattern, function in options.content_filters:
 		# Finding all matches...
 		text_tokens_index = 0
 		text_tokens_charpos = 0
 		text_tokens_token_xdiff = 0
 		text_content = "".join(t.value for t in text_tokens)
-		for m in pattern.finditer(text_content):
-			# We got a match at text_content[i1:i2].
-			i1 = m.start()
-			i2 = m.end()
-
-			# Pass the matched text to the replacement function to get replaced text.
-			replacement = function(m)
-
-			# Do a text replacement in the tokens that produced this text content.
-			# It may have been produced by multiple tokens, so loop until we find them all.
-			while i1 < i2:
-				# Find the original tokens in the content stream that
-				# produced the matched text. Start by advancing over any
-				# tokens that are entirely before this span of text.
-				while text_tokens_index < len(text_tokens) and \
-				      text_tokens_charpos + len(text_tokens[text_tokens_index].value)-text_tokens_token_xdiff <= i1:
-					text_tokens_charpos += len(text_tokens[text_tokens_index].value)-text_tokens_token_xdiff
-					text_tokens_index += 1
-					text_tokens_token_xdiff = 0
-				if text_tokens_index == len(text_tokens): break
-				assert(text_tokens_charpos <= i1)
-
-				# The token at text_tokens_index, and possibly subsequent ones,
-				# are responsible for this text. Replace the matched content
-				# here with replacement content.
-				tok = text_tokens[text_tokens_index]
-
-				# Where does this match begin within the token's text content?
-				mpos = i1 - text_tokens_charpos
-				assert mpos >= 0
-
-				# How long is the match within this token?
-				mlen = min(i2-i1, len(tok.value)-text_tokens_token_xdiff-mpos)
-				assert mlen >= 0
-
-				# How much should we replace here?
-				if mlen < (i2-i1):
-					# There will be more replaced later, so take the same number
-					# of characters from the replacement text.
-					r = replacement[:mlen]
-					replacement = replacement[mlen:]
-				else:
-					# This is the last token in which we'll replace text, so put
-					# all of the remaining replacement content here.
-					r = replacement
-					replacement = None # sanity
+		matches = [a for a in pattern.finditer(text_content)]
```

Comment:
we don't need the iterator `find` method if we're just going to unwrap it. I think `matches = pattern.findall(text_context)` should work equivalent

Reply:
ahh, good point :)

Reply:
Ahhh, I tried your suggestion and now I remember why I left the `finditer`. I tried `findall` before, too, but it returns a list of strings, whereas finditer gives  a colleciton of tuples which hold more information, like the position and length of the fragment of text.

Reply:
aahhh 👍